### PR TITLE
405 create new reward distribution pallet

### LINKED ISSUE #405: Create new reward-distribution pallet
This ticket is a part of https://github.com/pendulum-chain/tasks/issues/47

Create a new pallet from scratch that takes care of the collateral reward logic.

## Additional tasks
* Add a configuration parameter `Currency` of type `Currency<Self::AccountId>` (which is a reference to pallet-balances)
* Add a storage variable `RewardPerBlock` (of type `Balance`, using `OptionQuery`) that maintains the number of reward tokens minted per block – it is 0 by default
* Add a storage variable `RewardsAdaptedAt` (of type `BlockNumber`, using `OptionQuery`) that stores the active block number of the last time `RewardPerBlock` has been adapted
* Add a reference to the security pallet so that the reward-distribution pallet can use `active_block_number` and `parachain_block_expired` functions
* Add constant pallet configuration parameters `DecayInterval` (of type `BlockNumber`) and `DecayRate` (of type `Perquintill`).
* Add a root extrinsic that allows to set `RewardPerBlock`. It will also set `RewardsAdaptedAt` to the current active block number.

### PR DESCRIPTION
**Pull request by ebma**
This PR adds the basics for the `reward-distribution` pallet. For reviewing please read the detailed instructions contained in #405.

Closes #405.

### CODE REVIEW

**Comment by ebma** [comment]
Thanks for the review @gianfra-t. If you are happy with the changes, please submit an approval. Our repositories are configured such that they require at least 1 approving review before they can be merged. Besides that, the CI jobs need to succeed so that it can be merged but I cancelled it for now because I did not want it to clash with the other jobs, just FYI.

image

**Review by gianfra-t — APPROVED** [review]
Ready to merge!

**Review by b-yap — APPROVED** [review]

File: pallets/reward-distribution/src/lib.rs
Comment (ebma) [reply pallets/reward-distribution/src/lib.rs:73]: I want to keep it for now because we will extend it in the next two tickets.

File: pallets/reward-distribution/src/lib.rs
Comment (ebma) [reply pallets/reward-distribution/src/lib.rs:97]: I don't think we need it yet, let's decide later

**Comment by ebma** [comment]
I ran the tests locally and they passed. I want to merge this already to unblock the work that's building on top of this.

**Pull request merged**
